_posts/2025-04-09-#31664.md

```diff
@@ -10,17 +10,143 @@ status: upcoming
 commit:
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+### Motivation and Background
 
+The current fee estimator in Bitcoin Core, used by the wallet and exposed to clients via RPC, is [`CBlockPolicyEstimator`](https://github.com/bitcoin/bitcoin/blob/4c1906a500cacab385b09e780b54271b0addaf4b/src/policy/fees.h#L148).
 
-## Questions
+This estimator maintains a vector of fee rate buckets ranging from 1,000 sats/kvB to 10,000,000 sats/kvB. These buckets are spaced exponentially by a factor of 1.05. For example, if the first bucket is 1,000 sats/kvB, the next would be 1,050 sats/kvB, and so on.
+The estimator works by:
+- Tracking each transaction's fee rate and assigning it to the appropriate bucket.
+- Monitoring transactions as they leave the mempool. If a transaction is confirmed in a block, the estimator records success in the bucket, along with the number of blocks it took to prove (i.e., the difference between the confirmation block height and the height at which it was first seen in the mempool).
+- If a transaction is removed for any reason other than inclusion in a block, it is recorded as a failure.
+This data is aggregated using an exponentially decaying moving average, so old data points become less relevant over time.
+
+A naive fee rate estimate by this estimator would provide an estimate for a confirmation target n by going through these buckets from the lowest bucket to the highest bucket and returning the lowest fee rate bucket where more than 80% of the transactions first seen within these n blocks were confirmed.
+
+However, the fee estimator is more complex than this. It maintains three sets of these buckets: **Short-term**, **Medium-term**, and **Long-term**, each decaying at a different rate.
+
+These allow the estimator to provide fee estimates in two modes:
+- **Conservative** – Uses long-term data for more cautious estimates and is tailored toward users who cannot afford to RBF bump later, hence high fee rate estimates are often seen.
```

Comment:
I don't think it's an affordability thing, but more if wallets don't support it / they want to get it right on the first shot?

```suggestion
- **Conservative** – Uses long-term data for more cautious estimates and is tailored toward users who do not plan to not fee-bump. These fee rates tend to be higher.
```